src/System.IO.Pipes/src/System/IO/Pipes/PipeStream.Windows.cs

```diff
@@ -312,7 +312,7 @@ private Task WriteAsyncCorePrivate(byte[] buffer, int offset, int count, Cancell
             }
             else
             {
-                var completionSource = new PipeStreamCompletionSource(this, buffer, cancellationToken, isWrite: true);
+                var completionSource = new PipeIOCompletionSource(this, buffer, cancellationToken, isWrite: true);
```

Comment:
Reading through this again, I still think ReadWriteCompletionSource would be a better name.  Then we have ConnectionCompletionSource for connecting, ReadWriteCompletionSource for reading/writing, and PipeCompletionSource for the base.